### DISCUSSION TITLE
bug:  Failed Requests On Android Simulator

### DISCUSSION THREAD

**Discussion by micahosborne** [ionic-team/capacitor#3672] [Q&A]
Latest Dependencies:

  @capacitor/cli: 2.4.2
  @capacitor/core: 2.4.2
  @capacitor/android: 2.4.2
  @capacitor/electron: 2.4.2
  @capacitor/ios: 2.4.2

Installed Dependencies:

  @capacitor/cli 2.4.2
  @capacitor/android 2.4.2
  @capacitor/core 2.4.2
  @capacitor/ios 2.4.2
  @capacitor/electron not installed

[success] Android looking great! 👌
  Found 10 Capacitor plugins for ios:
    cordova-plugin-androidx (1.0.2)
    cordova-plugin-androidx-adapter (1.1.1)
    cordova-plugin-device (2.0.3)
    cordova-plugin-ionic (5.4.7)
    cordova-plugin-ionic-keyboard (2.2.0)
    cordova-plugin-ionic-webview (4.2.1)
    cordova-plugin-splashscreen (5.0.4)
    cordova-plugin-statusbar (2.4.3)
    cordova-plugin-whitelist (1.3.4)
    cordova-sqlite-storage (3.3.0)
[success] iOS looking great! 👌

I'm trying to post a request using Angular/Ionic httpclient.  It works fine in iOS and browser, but on android emulator it automatically cancels my request.  I've googled and tried literally everything. 

Seriously how is this even a product, surely every capacitor app performs XHR requests to other domains? I get "cancelled" under chrome network tab. I've tried "allowNaviigation,...etc" and every other combination in capacitor.config.json 

It halts my entire app from doing anything.  I"ve spent 2 days on this.. Can't believe i'm the first to post

Also, to note, this worked perfectly fine with cordova, as soon as i moved to capacitor it now fails.

**Comment by jcesarmobile** [comment]
XHR and fetch work perfectly fine on Capacitor.
But there are a few common problems users can hit.

1. Not having CORS correctly configured
2. Using http urls instead of https. Since Android 9, the OS blocks http calls by default. To be able to use http you have enable cleartext traffic.
3. Using bad/untrusted/self signed certificates. If using https but the device doesn’t trust the certificates.
4. Wrong time/date on the testing device. This can cause calls to https endpoints to fail.
5. Trying to connect to localhost. On Android simulator or real devices localhost is the device/emulator, not the server where you are testing, so it fails. It still works on browser and iOS simulator since they are running on the same computer as the server.

Those are the more common, there are a few more. If you provide more information then maybe you can get better help.

**Reply by simonflk** [reply]
Thank you -- I ran into this today and I didn't know about the time/date issue. You saved me several hours 🙏🏼
